backend/api/Controllers/AuthController.cs

```diff
@@ -54,8 +54,8 @@ public AuthController(IOptionsMonitor<Keycloak.Configuration.KeycloakOptions> op
         [SwaggerOperation(Tags = new[] { "auth" })]
         public IActionResult Activate()
         {
-            var keycloakUserId = this.User.GetGuid();
-            var exists = _pimsService.User.UserExists(keycloakUserId);
+            string username = this.User.GetUsername();
```

Comment:
You'll need to review the `this.User.GetUsername()` extension method.  If it simply returns the value from the JWT it will not match the username in the DB.